### ISSUE TITLE
Iterable feature

### ISSUE THREAD

**Issue by janajri** [vhf/bplustree#2]
Wondering if there are any plans to add an iterable. Could use in this way

```
const limit = 100
const min = 'aKJshiId'
const iter = tree.entries({ min })
const results = []
while(!iter.done && results.length < limit) {
   results.push(iter.next())
}
return results
```

**Comment by vhf** [comment]
Why not, yeah. Could you elaborate on how you would like to use this?

What is `min` in your example?

**Comment by janajri** [comment]
My use case would be paging data out of the store. 'min' could be an optional key to begin reading out of in either ascending or descending order. The `fetchRange` method requires a maximum but this would allow a min/limit or skip/limit is perhaps more appropriate assuming one direction.

**Comment by janajri** [comment]
Another solution for my specific use case could be treating 'max' as optional, but also adding a 'limit' parameter to cut the reads short.

**Comment by vhf** [comment]
Ok. To clarify, you'd like all the following:
- to go through the leafs in `key` order, either going 'left' or 'right'
- setting a limit number of leafs to go through
- or setting a 'target' `value`, at which iteration stops.

Is that right? Did I get the last one wrong, and you'd rather have a way of stopping at a target `key`?

**Comment by janajri** [comment]
That is correct. I would like to stop after I have read 'x' number of items from the tree, so as not to traverse more than what is necessary.

**Comment by vhf** [comment]
Here is a proof of concept @janajri : https://github.com/vhf/bplustree/blob/98da4ccdf8a2994286af0a6b5ba0e7c370a74f9a/lib/bplustree.js#L143-L182

You can check this branch if you want to test the API locally: https://github.com/vhf/bplustree/tree/values-generator.

I'll add tests later and push as soon as I'm sure it fits your needs.

**Comment by vhf** [comment]
Published as 1.2.0

**Comment by janajri** [comment]
Great! Thanks for that quick turnaround.
